pkg/apis/extensions/helpers.go

```diff
@@ -34,13 +33,13 @@ func PodSelectorAsSelector(ps *PodSelector) (labels.Selector, error) {
 	if len(ps.MatchLabels)+len(ps.MatchExpressions) == 0 {
 		return labels.Everything(), nil
 	}
-	selector := labels.LabelSelector{}
+	selector := labels.NewSelector()
 	for k, v := range ps.MatchLabels {
-		req, err := labels.NewRequirement(k, labels.InOperator, sets.NewString(v))
+		r, err := labels.NewRequirement(k, labels.InOperator, sets.NewString(v))
 		if err != nil {
 			return nil, err
 		}
-		selector = append(selector, *req)
+		selector = selector.Add(*r)
```

Comment:
It's weird that NewRequirement returns a pointer, and this is going to call sort.Sort a bunch but I guess it's fine.